wpilibNewCommands/src/test/native/cpp/frc2/command/DeferredCommandTest.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "CommandTestBase.h"
+#include "frc2/command/Commands.h"
+#include "frc2/command/DeferredCommand.h"
+
+using namespace frc2;
+
+class DeferredCommandTest : public CommandTestBase {
+ public:
+  void DeferredFunctionsTest(bool interrupted) {
+    std::unique_ptr<MockCommand> innerHolder = std::make_unique<MockCommand>();
+    MockCommand* innerCommand = innerHolder.get();
+    DeferredCommand deferred{[innerCommand] { return innerCommand; }, {}};
+
+    EXPECT_CALL(*innerCommand, Initialize());
+    EXPECT_CALL(*innerCommand, Execute());
+    EXPECT_CALL(*innerCommand, End(interrupted));
+
+    deferred.Initialize();
+    deferred.Execute();
+    EXPECT_FALSE(deferred.IsFinished());
+    innerCommand->SetFinished(true);
+    EXPECT_TRUE(deferred.IsFinished());
+    deferred.End(interrupted);
+  }
+};
+
+TEST_F(DeferredCommandTest, DeferredFunctions) {
+  DeferredFunctionsTest(false);
+  DeferredFunctionsTest(true);
```

Comment:
Ensure it's obvious which of the two failed.
[Parameterized tests](https://google.github.io/googletest/advanced.html#value-parameterized-tests) feel like the correct solution; [`SCOPED_TRACE`](https://google.github.io/googletest/advanced.html#using-assertions-in-sub-routines) is also an option (and I'm sure there are others).